File: visible-ios/visible/visible/SharingLogic.swift
import Foundation

/// The pure role-to-label derivation behind ``SharingModel``, with no
/// `AppHandle` or `@Observable` state. The model delegates here so the labels
/// are exercised directly.
enum SharingLogic {
    /// The label for a role: the row label in the members list and the option
    /// label in the invite picker.
    static func roleName(_ role: BridgeMemberRole) -> String {
        switch role {
        case .owner: "Owner"
        case .member: "Member"
        case .follower: "Follower"
        }
    }
}

File: visible-android/app/src/main/java/fm/bae/visible/ui/SharingLogic.kt
package fm.bae.visible.ui

import uniffi.visible_bridge.BridgeMemberRole

/**
 * The pure role-to-label derivation behind [SharingViewModel], with no
 * [uniffi.visible_bridge.AppHandle] or observable state. The view model
 * delegates here so the labels are exercised directly.
 */
object SharingLogic {
    /**
     * The label for a role: the row label in the members list and the option
     * label in the invite picker.
     */
    fun roleName(role: BridgeMemberRole): String = when (role) {
        BridgeMemberRole.OWNER -> "Owner"
        BridgeMemberRole.MEMBER -> "Member"
        BridgeMemberRole.FOLLOWER -> "Follower"
    }
}
